scripts/japicmp.sh

```diff
@@ -52,9 +86,19 @@ if [ -z "${OLD_ST_VERSION}" ]; then
   exit 1
 fi
 
+OLD_ARTIFACTS="$(bom_artifacts "${GROUP_ID}" "${OLD_ST_VERSION}")"
+
+if [ "${LOCAL}" = "local" ]; then
+  NEW_ARTIFACTS="$(find servicetalk-* -type d -maxdepth 0 | sort -)"
+else
+  NEW_ARTIFACTS="$(bom_artifacts "${GROUP_ID}" "${NEW_ST_VERSION}")"
+fi
+
 # All servicetalk modules except:
 # servicetalk-benchmarks, servicetalk-bom, servicetalk-examples, servicetalk-gradle-plugin-internal
-ARTIFACTS="$(find servicetalk-* -type d -maxdepth 0 |
+ARTIFACTS="$(comm -1 -2 \
+  <(echo "${OLD_ARTIFACTS}" | tr ' ' '\n') \
+  <(echo "${NEW_ARTIFACTS}" | tr ' ' '\n') |
```

Comment:
Do we need `NEW_ARTIFACTS` in the final `ARTIFACTS`? We should only make sure that old artifacts are compatible. If new artifacts were added, there is nothing we can compare with from `OLD_ARTIFACTS`

Reply:
`comm` is generating a list that is the intersection of the old and new artifacts. If either old or new contains artifacts that the other does not then they will not be compared.

Reply:
Oh, I missed `comm` at the beginning, thanks for clarifying!